pkg/operation/botanist/controlplane.go

```diff
@@ -206,3 +222,23 @@ func (b *Botanist) RestartControlPlanePods(ctx context.Context) error {
 		client.MatchingLabels{v1beta1constants.LabelPodMaintenanceRestart: "true"},
 	)
 }
+
+func waitUntilPodsAreTerminatedForDeployments(ctx context.Context, c client.Client, namespace string, deployments []string) <-chan error {
```

Comment:
Can't we use the `Parallel(...)` function from the flow package here: https://github.com/gardener/gardener/blob/bbe19d1c11c59b7a595decfde227947725f4d7f4/pkg/utils/flow/taskfn.go#L119